src/main/kotlin/no/nav/syfo/personstatus/PersonTildelingService.kt

```diff
@@ -1,11 +1,30 @@
 package no.nav.syfo.personstatus
 
-import no.nav.syfo.personstatus.infrastructure.database.DatabaseInterface
-import no.nav.syfo.personstatus.db.lagreVeilederForBruker
+import no.nav.syfo.personstatus.application.IPersonOversiktStatusRepository
+import no.nav.syfo.personstatus.domain.PersonIdent
 import no.nav.syfo.personstatus.domain.VeilederBrukerKnytning
+import no.nav.syfo.personstatus.infrastructure.cronjob.behandlendeenhet.PersonBehandlendeEnhetService
 
-class PersonTildelingService(private val database: DatabaseInterface) {
-
-    fun lagreKnytningMellomVeilederOgBruker(veilederBrukerKnytninger: List<VeilederBrukerKnytning>) =
-        veilederBrukerKnytninger.map { database.lagreVeilederForBruker(it) }
+class PersonTildelingService(
+    private val personoversiktStatusRepository: IPersonOversiktStatusRepository,
+    private val personBehandlendeEnhetService: PersonBehandlendeEnhetService,
+) {
+    suspend fun lagreKnytningMellomVeilederOgBruker(
+        veilederBrukerKnytninger: List<VeilederBrukerKnytning>,
+        tildeltAv: String,
+    ) {
+        veilederBrukerKnytninger.forEach {
+            val personident = PersonIdent(it.fnr)
+            val personoversiktStatus = personoversiktStatusRepository.getOrCreatePersonOversiktStatusIfMissing(
+                personident = personident,
+            )
```

Comment:
Hadde dette gjort det litt enklere å lese? Så hadde vi også sluppet logikk inne i repositoriet 😊
```suggestion
            val personoversiktStatus = personoversiktStatusRepository.getPersonOversiktStatus(personident = personident) ?: personoversiktStatusRepository.createPersonOversiktStatus(personident)
```

Reply:
Ja, eneste ulempen er at vi ikke får gjort det med en og samme connection. Men det feiler vel evnt på en unique constraint, så enig i at dette blir bedre 👍